Source/BuildControl.cs

```diff
@@ -718,16 +718,22 @@ public void LoadCraft (string filename, string flagname)
 				Debug.LogWarning ($"File '{filename}' does not exist");
 				return;
 			}
-			string craftText = File.ReadAllText (filename);
-			ConfigNode craft = ConfigNode.Parse (craftText);
-			ReplaceLaunchClamps (craft);
 
-			state = State.Planning;
-			craftName = Localizer.Format (craft.GetValue ("ship"));
-			if ((buildCost = getBuildCost (craft, craftText)) != null) {
-				craftConfig = craft;
+			try {
+				string craftText = File.ReadAllText(filename);
+				ConfigNode craft = ConfigNode.Parse(craftText);
+				ReplaceLaunchClamps(craft);
+
+				state = State.Planning;
+				craftName = Localizer.Format(craft.GetValue("ship"));
+				if ((buildCost = getBuildCost(craft, craftText)) != null) {
+					craftConfig = craft;
+				}
+				PlaceCraftHull();
+			} catch (Exception e) {
+				Debug.LogException(e);
+				state = State.Idle;
 			}
-			PlaceCraftHull ();
```

Comment:
I would prefer to not have a try/catch at all. First, whatever mods cause exceptions really need to be fixed. Then, really, far too much is covered by the try block. Finally, the real problem is that the setting of `state` and `craftName` should come /after/ the call to `getBuildCost` (maybe inside the conditional block, undecided) rather than before the call.